### PR TITLE
Minor linux improvements

### PR DESCRIPTION
**Pull request by Gloorf**
Just a couple of improvements for the dozens of people using acquisition on linux

### CODE REVIEW

File: src/currencymanager.cpp [hunk 1]
```diff
@@ -237,6 +237,9 @@ CurrencyDialog::CurrencyDialog(CurrencyManager& manager) : currency_manager_(man
     UpdateTotalWisdomValue();
     layout_->addWidget(new QLabel("Total Scrolls of Wisdom"), curr_row + 1, 0);
     layout_->addWidget(total_wisdom_value_, curr_row + 1, 1);
+    button_close_ = new QPushButton("Close");
```
Comment (xyzz) [inline src/currencymanager.cpp]: is there no close button on the top left/right on linux?

Comment (Gloorf) [inline src/currencymanager.cpp]: Yep, see http://imgur.com/29AwgrA
I could also override the windowFlags, but that could lead to unexpected behaviour on other OS/WM. This way it's non intrusive and works everywhere

File: src/mainwindow.cpp [hunk 2]
```diff
@@ -262,7 +265,10 @@ void MainWindow::OnStatusUpdate(const CurrentStatusUpdate &status) {
     progress->setMaximum(status.total);
     progress->setValue(status.progress);
     progress->setPaused(status.state == ProgramState::ItemsPaused);
+#else
+    if (need_progress) {;}//Fix compilation warning
```
Comment (xyzz) [inline src/mainwindow.cpp]: just use `(void)need_progress;` here instead of the `if`

File: src/shop.cpp [hunk 3]
```diff
@@ -151,7 +151,7 @@ std::string Shop::ShopEditUrl(int idx) {
 }
 
 void Shop::SubmitSingleShop() {
-    CurrentStatusUpdate status = {};
+    CurrentStatusUpdate status = {ProgramState::ShopSubmitting, 0, 0};
```
Comment (xyzz) [inline src/shop.cpp]: does `CurrentStatusUpdate status = CurrentStatusUpdate();` get rid of the warning?

File: src/currencymanager.cpp (hunk 1 above)
Comment (xyzz) [inline src/currencymanager.cpp]: I think it would be better to override `windowFlags`. What unexpected behavior could happen?

Comment (Gloorf) [inline src/currencymanager.cpp]: I don't know, it is unexpected :P Probably nothing weird though, changed it

File: src/datamanager.cpp [hunk 4]
```diff
@@ -85,7 +85,7 @@ std::vector<CurrencyUpdate> DataManager::GetAllCurrency() {
     sqlite3_prepare(db_, query.c_str(), -1, &stmt, 0);
     std::vector<CurrencyUpdate> result;
     while (sqlite3_step(stmt) == SQLITE_ROW) {
-        CurrencyUpdate update = { 0 };
+        CurrencyUpdate update = { 0, "" };
```
Comment (xyzz) [inline src/datamanager.cpp]: forgot one

**Comment by xyzz** [comment]
pushed as 822a1fbc71472063cf0793347bc467e2d69945ab

**Pull request closed without merging**